src/Collector.cs

```diff
@@ -142,14 +136,14 @@ private void SendQueuedMessages(Socket socket)
         }
     }
 
-    private bool Authenticate(Socket socket)
+    private void Authenticate(Socket socket)
     {
       var data = System.Text.Encoding.ASCII.GetBytes("hello version dotnet/0.2.0\n");
       socket.Send(data);
-      if(!ReceiveOk(socket)) return false;
+      if(!ReceiveOk(socket)) throw new Exception("Instrumental Authentication Failed");
       data = System.Text.Encoding.ASCII.GetBytes($"authenticate {_apiKey}\n");
       socket.Send(data);
-      return ReceiveOk(socket);
+      if(!ReceiveOk(socket)) throw new Exception("Instrumental AuthenticationFailed");
```

Comment:
By moving line 143 next to 146 we should be able to eliminate one round trip from the connect without a change in protocol.

P.S. I think we're missing a space in the message on this line.